plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/expression/SparkExpressionBuilder.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.expression;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static java.lang.String.format;
+import static java.util.HexFormat.isHexDigit;
+
+public class SparkExpressionBuilder
+        extends SparkExpressionBaseVisitor<Object>
+{
+    private static final char SPARK_ESCAPE_CHARACTER = '\\';
+
+    @Override
+    public Object visitStandaloneExpression(SparkExpressionParser.StandaloneExpressionContext context)
+    {
+        return visit(context.expression());
+    }
+
+    @Override
+    public Object visitPredicated(SparkExpressionParser.PredicatedContext context)
+    {
+        // Handle comparison operator
+        if (context.predicate() != null) {
+            return visit(context.predicate());
+        }
+
+        // Handle simple expression likes just TRUE
+        return visit(context.valueExpression);
+    }
+
+    @Override
+    public Object visitComparison(SparkExpressionParser.ComparisonContext context)
+    {
+        return new ComparisonExpression(
+                getComparisonOperator(((TerminalNode) context.comparisonOperator().getChild(0)).getSymbol()),
+                (SparkExpression) visit(context.value),
+                (SparkExpression) visit(context.right));
+    }
+
+    @Override
+    public SparkExpression visitAnd(SparkExpressionParser.AndContext context)
+    {
+        verify(context.booleanExpression().size() == 2, "AND operator expects two expressions: " + context.booleanExpression());
+        return new LogicalExpression(LogicalExpression.Operator.AND, visit(context.booleanExpression(0), SparkExpression.class), visit(context.booleanExpression(1), SparkExpression.class));
```

Comment:
- Split the call into lines

- in the grammar you can name the left and right expressions: `| left=booleanExpression AND right=booleanExpression` and then refer to them by name